### PR TITLE
Fix/improve support for the BFD library

### PR DESCRIPTION
**Pull request by shindere**
This pull request fixes and improves the support for the BFD
(Binary File Description) library.

It adds the `--with-bfd` configure-time option to request that the
support is present. If this option is given and the library is not found,
that will result in a configure error.

Two configure variables, namely `BFD_INCLUDE_DIR` and `BFD_LIB_DIR` are
also provided to let users specify where the BFD header files and
libraries are located.

The detection of BFD on MacOSX is also fixed. It has actually been
broken for several versions of OCaml, even before the transition to
autoconf. For the record, on MacOSX the BFD library is provided by
MacPorts' binutils package.

This PR is probably best reviewed commit by commit. Also, the names of
the BFD-related build variables have been modified to bring them closer to
GNU conventions. Similar changes will be proposed soon for other
names of build variables.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1455,30 +1468,61 @@ AS_IF([test x"$enable_systhreads" = "xno"],
...
+    AS_IF([test -n "$bfd_ldlibs"],
+      [bfd_available=true
+      AC_DEFINE([HAS_LIBBFD])])])
+  AS_IF([! $bfd_available],
+    [AS_IF([test x"$with_bfd" = "xyes"],
+      [AC_MSG_ERROR([BFD library support requested but not available])],
+      [bfd_cppflags=""
+      bfd_ldflags=""
+      AC_MSG_NOTICE(m4_normalize([
+        BFD library not found, 'objinfo' will be unable to display
```
Comment (damiendoligez) [inline configure.ac]:
```suggestion
        BFD library not found, 'ocamlobjinfo' will be unable to display
```

File: configure.ac [hunk 2]
```diff
@@ -1455,30 +1468,61 @@ AS_IF([test x"$enable_systhreads" = "xno"],
...
+        info on .cmxs files.
+      ]))])])
+  LDFLAGS="$SAVED_LDFLAGS"
+  CPP_FLAGS="$SAVED_CPPFLAGS"],
+  [AC_MSG_NOTICE(m4_normalize([
+    Support for the BFD (Binary File Description) library disalbed,
```
Comment (damiendoligez) [inline configure.ac]:
```suggestion
    Support for the BFD (Binary File Description) library disabled,
```

File: configure.ac [hunk 3]
```diff
@@ -1455,30 +1468,61 @@ AS_IF([test x"$enable_systhreads" = "xno"],
...
+    'objinfo' will be unable to display info on .cmxs files.
```
Comment (damiendoligez) [inline configure.ac]:
```suggestion
    'ocamlobjinfo' will be unable to display info on .cmxs files.
```

**Review by damiendoligez — APPROVED** [review]
LGTM. Only a few small things to change.

**Comment by shindere** [comment]
Thanks. I rebased, integrated the changes, updated the change entry and
merged without waiting for CI because all checks passed before.

**Pull request merged**